packages/flutter_tools/lib/executable.dart

```diff
@@ -86,9 +86,7 @@ Future<void> main(List<String> args) async {
   final bool muteCommandLogging = (help || doctor) && !veryVerbose;
   final bool verboseHelp = help && verbose;
   final bool daemon = args.contains('daemon');
-  final bool runMachine =
-      (args.contains('--machine') && args.contains('run')) ||
-      (args.contains('--machine') && args.contains('attach'));
+  final bool runMachine = args.contains('--machine');
```

Comment:
lol. much nicer, wtf

Reply:
Right? I don't even know why we'd want to limit this if we have multiple commands that support `--machine` and we create the logger factory here...